main.tf

```diff
@@ -0,0 +1,84 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 2.49"
+    }
+  }
+
+  backend "azurerm" {
+    resource_group_name  = "SoftwirePilot_JoelWee_ProjectExercise"
+    storage_account_name = "joelweetfstate"
+    container_name       = "tfstate-container"
+    key                  = "terraform.tfstate"
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name = "${var.prefix}_ProjectExercise"
+}
+
+resource "azurerm_app_service_plan" "main" {
+  name                = "terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  kind                = "Linux"
+  reserved            = true
+  sku {
+    tier = "Basic"
+    size = "B1"
+  }
+}
+resource "azurerm_app_service" "main" {
+  name                = "terraformed-app"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  app_service_plan_id = azurerm_app_service_plan.main.id
+  site_config {
+    app_command_line = ""
+    linux_fx_version = "DOCKER|j0elwee/devops-trg:latest"
+  }
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+    "MONGO_URI"                  = "mongodb://${azurerm_cosmosdb_account.main.name}:${azurerm_cosmosdb_account.main.primary_key}@${azurerm_cosmosdb_account.main.name}.mongo.cosmos.azure.com:10255/DefaultDatabase?ssl=true&replicaSet=globaldb&retrywrites=false&maxIdleTimeMS=120000"
+    "OAUTH_CLIENT_ID"            = "${var.oauth_client_id}"
+    "OAUTH_CLIENT_SECRET"        = "${var.oauth_client_secret}"
+    "SECRET_KEY"                 = "${var.secret_key}"
```

Comment:
Minor, but you can just use the var directly rather than via interpolation; your travis pipeline shows a warning on this:

```
Warning: Interpolation-only expressions are deprecated
  on main.tf line 48, in resource "azurerm_app_service" "main":
  48:     "OAUTH_CLIENT_ID"            = "${var.oauth_client_id}"
Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.
Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.
```